desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.desktop.components;
+
+import bisq.desktop.util.GUIUtil;
+
+import bisq.core.locale.Res;
+import bisq.core.user.BlockChainExplorer;
+import bisq.core.user.Preferences;
+
+import bisq.common.util.Utilities;
+
+import de.jensd.fx.fontawesome.AwesomeDude;
+import de.jensd.fx.fontawesome.AwesomeIcon;
+
+import com.jfoenix.controls.JFXTextField;
+
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.control.Tooltip;
+import javafx.scene.layout.AnchorPane;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.annotation.Nullable;
+
+public class ExplorerAddressTextField extends AnchorPane {
+    private static Preferences preferences;
+
+    public static void setPreferences(Preferences preferences) {
+        ExplorerAddressTextField.preferences = preferences;
+    }
```

Comment:
setters should be further down the file, after variable declarations and constructors

Reply:
Or annotate `preferences` with `@Setter` and delete this method

Reply:
`preferences` is static field, not sure why - I used same approach like in `TxIdTextField`

Reply:
> `preferences` is static field, not sure why - I used same approach like in `TxIdTextField`
Yes it is true that it is right now the same as in TxIdTextField, but I also would prefer using the @Setter annotation and getting rid of the unnecessary setPreferences method.

Reply:
True - I thought it's not possible to use Setter on static field. Fixed it here and also in TxIdTextField